pkg/controller/directimagemigration/directimagemigration_controller.go

```diff
@@ -70,18 +73,20 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	err = c.Watch(
 		&source.Kind{Type: &migapi.MigCluster{}},
 		handler.EnqueueRequestsFromMapFunc(func(a client.Object) []reconcile.Request {
-			return migref.GetRequests(a, migapi.DirectImageMigration{})
+			return migref.GetRequests(a, migapi.OpenshiftMigrationNamespace, migapi.DirectImageMigration{})
 		}),
+		&migref.MigrationNamespacePredicate{Namespace: migapi.OpenshiftMigrationNamespace},
 	)
 	if err != nil {
 		return err
 	}
 
 	// Watch for changes to DirectImageStreamMigrations
-	err = c.Watch(&source.Kind{Type: &migapi.DirectImageStreamMigration{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &migapi.DirectImageMigration{},
-	})
+	err = c.Watch(&source.Kind{Type: &migapi.DirectImageStreamMigration{}},
+		&handler.EnqueueRequestForOwner{
+			IsController: true,
+			OwnerType:    &migapi.DirectImageMigration{},
+		}, &migref.MigrationNamespacePredicate{Namespace: migapi.OpenshiftMigrationNamespace})
```

Comment:
pointer is being used here `&migref.MigrationNamespacePredicate{Namespace: migapi.OpenshiftMigrationNamespace}` but the predicate methods are on non-pointer object `func (m MigrationNamespacePredicate) Create(evt event.CreateEvent) bool {
	if evt.Object.GetNamespace() == m.Namespace {
		return true
	}
	return false
}
`

Not sure what effect this will have

Reply:
@alaypatel07 wont have any effect right? Because we are only consuming the `Namespace` field in the struct which itself isn't a pointer. managers only call the function on the struct and pass deps (events) from outside

Reply:
yeah, I am still confused about how it works, will have to dig into the type system. But it works so going to approve it.